File: languages/go/entity/encryption_alg.go
package entity

import "github.com/bloock/bloock-sdk-go/v2/internal/bridge/proto"

type EncryptionAlg int32

const (
	AES256GCM                   EncryptionAlg = iota
	RSA                         EncryptionAlg = iota
	ECIES                       EncryptionAlg = iota
	UNRECOGNIZED_ENCRYPTION_ALG EncryptionAlg = -1
)

var (
	EncryptionAlgFromProto = map[proto.EncryptionAlg]EncryptionAlg{
		0: AES256GCM,
		1: RSA,
		2: ECIES,
	}
	EncryptionAlgToProto = map[EncryptionAlg]proto.EncryptionAlg{
		AES256GCM: 0,
		RSA:       1,
		ECIES:     2,
	}
)

File: languages/js/src/entity/encryption_alg.ts
import * as proto from "../bridge/proto/encryption_entities";

export enum EncryptionAlg {
  AES256GCM = 0,
  RSA = 1,
  ECIES = 2,
  UNRECOGNIZED = -1
}

export namespace EncryptionAlg {
  export function toProto(alg: EncryptionAlg): proto.EncryptionAlg {
    switch (alg) {
      case EncryptionAlg.AES256GCM:
        return proto.EncryptionAlg.A256GCM;
      case EncryptionAlg.RSA:
        return proto.EncryptionAlg.RSA;
      case EncryptionAlg.ECIES:
        return proto.EncryptionAlg.ECIES;
      case EncryptionAlg.UNRECOGNIZED:
        return proto.EncryptionAlg.UNRECOGNIZED;
    }
  }

  export function fromProto(
    alg: proto.EncryptionAlg | undefined
  ): EncryptionAlg {
    switch (alg) {
      case proto.EncryptionAlg.A256GCM:
        return EncryptionAlg.AES256GCM;
      case proto.EncryptionAlg.RSA:
        return EncryptionAlg.RSA;
      case proto.EncryptionAlg.ECIES:
        return EncryptionAlg.ECIES;
      case proto.EncryptionAlg.UNRECOGNIZED:
      case undefined:
        return EncryptionAlg.UNRECOGNIZED;
    }
  }
}
